### PR TITLE
[UI] Registry: Perform model fetch on toggle and InfiniteScroll

### PR DESCRIPTION
**Pull request by dottharun**
**Notes for Reviewers**

This PR fixes 
### settings/registry models tab:
- and only fetches when the version dropdown tree is toggled.
- lazy-loading of pages with `IntersectionObserver`.

### registrants tab:
- Observer added.

### Note:
- Aggregation of models is still done in UI for registrants - removing this requires fixing duplication of data from the API. (but can be done separately with removal of all duplications in the registry)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Yashsharma1911** [comment]
Can you add video of change you made? on clicking on version we do show the loader right? incase API have some latency otherwise it will look like buggy

**Comment by Yashsharma1911** [comment]
Registry supports deep linking are we making sure after this change it will be working?

File: ui/components/MeshModelRegistry/MesheryTreeView.js [hunk 1]
```diff
@@ -176,70 +196,87 @@ const MesheryTreeViewItem = ({
             }
             check={true}
             onClick={() => {
+              getModelByNameTrigger({
+                name: versionedModelDef.name,
+                params: {
+                  version: versionedModelDef.model.version,
+                  components: true,
+                  relationships: true,
+                },
```
Comment (Yashsharma1911) [inline ui/components/MeshModelRegistry/MesheryTreeView.js:205]: By default lazy loading does not use caching, please add this prop to make sure it is using it's caching behaviour
```suggestion
                }, true
```

File: ui/components/MeshModelRegistry/MesheryTreeView.js [hunk 2]
```diff
@@ -176,70 +196,87 @@ const MesheryTreeViewItem = ({
...
+                    view={MODELS}
+                    setShowDetailsData={setShowDetailsData}
+                    idForKindAsProp={`${registrantID ? `${registrantID}.1.` : ''}${modelDef.id}.${
+                      selectedVersionedModel.id
+                    }.2`}
+                  />
+                </StyledTreeItem>
+              </>
+            ) : (
+              <CircularProgress />
```
Comment (Yashsharma1911) [inline ui/components/MeshModelRegistry/MesheryTreeView.js:278]: Okay you already added here thanks, please add this prop to make sure it gets update with theme, also please test after update
```suggestion
              <CircularProgress color="inherit"/>
```

**Comment by leecalcote** [comment]
Suggested code changes applied.

**Comment by Yashsharma1911** [comment]
In video on first item we are not showing the name, I think it is because that model doesn't have displayName can you add a fallback if displayName is not present we by default show the "name" of model

**Comment by Yashsharma1911** [comment]
Testing

**Comment by Yashsharma1911** [comment]
https://github.com/user-attachments/assets/b76e2995-7923-4ab4-a3ed-ddc887eb8d74

@dottharun after scrolling few time in models tab and then switching to other tab and switch back to models, models gets load from page where I left before switching and causing issue of losing prev models info and there is no way I can fetch prev data

**Comment by Yashsharma1911** [comment]
After switching just make sure we are clearing page state

**Comment by dottharun** [comment]
@Yashsharma1911 Made the changes here.

**Comment by Yashsharma1911** [comment]
Thankyou testing

**Comment by leecalcote** [comment]
Is this the only bug left, @Yashsharma1911? You didn't find any others?

**Comment by dottharun** [comment]
test

**Comment by dottharun** [comment]
test

**Pull request closed without merging**